src/lvl_script.c

```diff
@@ -6592,6 +6606,12 @@ void script_process_value(unsigned long var_index, unsigned long plr_range_id, l
     }
     break;
   }
+  case Cmd_RANDOMISE_FLAG:
+      for (i=plr_start; i < plr_end; i++)
+      {
+          set_script_flag(i,val2,saturate_set_unsigned(rand() % (val3 + 1), 8));
```

Comment:
It sets just the regular old flag, do you think you can use it to set all flag types?
https://github.com/dkfans/keeperfx/wiki/New-and-Modified-Level-Script-Commands#set_flag

the ADD_TO_FLAG command does, perhaps you can steal the code from there.